File: app/src/main/java/noor/serry/alram/workManager/RingAlarm.java
package noor.serry.alram.workManager;

import android.app.backup.SharedPreferencesBackupHelper;
import android.content.Context;
import android.content.SharedPreferences;
import android.media.MediaPlayer;
import android.media.RingtoneManager;
import android.net.Uri;
import android.os.Environment;
import android.os.Handler;

import androidx.annotation.NonNull;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

import noor.serry.alram.R;

public class RingAlarm extends Worker {
    Context context;
    MediaPlayer mediaPlayer;
    Uri musicUri ;
    public RingAlarm(@NonNull Context context, @NonNull WorkerParameters workerParams) {
        super (context, workerParams);
        this.context=context;
    }


    @NonNull
    @Override
    public Result doWork() {
        creatMediaPlayer ();
        mediaPlayer.start ();
        sleep (30);
        mediaPlayer.stop ();
        return Result.success ();
    }

        private void creatMediaPlayer(){
        mediaPlayer =  MediaPlayer.create(context,getUri ());
    }

        private Uri getUri(){
        setMusicUri ();
        return musicUri;
    }

        private void setMusicUri(){
        int musicIndex = RingtoneManager.TYPE_RINGTONE;
        musicUri = RingtoneManager.getDefaultUri (musicIndex);
    }
        private void sleep(int seconds)  {
            try {
                Thread.sleep (seconds*1000);
            } catch (InterruptedException e) {
                e.printStackTrace ( );
            }
    }
}

File: app/src/main/java/noor/serry/alram/util/workManager/RingAlarm.kt
package noor.serry.alram.util.workManager

import android.content.Context
import androidx.work.WorkerParameters
import android.media.MediaPlayer
import android.media.RingtoneManager
import android.net.Uri
import android.widget.Toast
import androidx.work.Worker
import kotlinx.coroutines.*
import noor.serry.alram.data.Repository

class RingAlarm(var context: Context, workerParams: WorkerParameters) : Worker(context, workerParams)
{
    var mediaPlayer: MediaPlayer? = null
    var musicUri: Uri? = null
    val timeRang = 30
    val  repository = Repository(context)


    override fun doWork(): Result {
        GlobalScope.launch (newSingleThreadContext("ringThread")){
            creatMediaPlayer()
            mediaPlayer!!.start()
            sleep(timeRang)
            mediaPlayer!!.stop()
            updateDataAfterRing()
           withContext(Dispatchers.Main){
               Toast.makeText(context,"done",Toast.LENGTH_LONG).show()
           }
       }
        return Result.success()
    }

    private fun creatMediaPlayer() {
        mediaPlayer = MediaPlayer.create(context, uri)
    }

    private val uri: Uri?
        private get():Uri?{
            setMusicUri()
            return musicUri
        }

    private fun setMusicUri() {
        val musicIndex = RingtoneManager.TYPE_RINGTONE
        musicUri = RingtoneManager.getDefaultUri(musicIndex)
    }

    private  fun sleep(seconds: Int) {
        try {
            Thread.sleep((seconds * 1000).toLong())
        } catch (e: InterruptedException) {
            e.printStackTrace()
        }
    }

    private fun updateDataAfterRing(){
        var tag = inputData.getString("tag")
        repository.updateAlarm(Integer.parseInt(tag),false)

    }
}